lib/oc-tests/test_shareFile.py

```diff
@@ -176,7 +176,7 @@ def shareeOne(step):
     list_files(d)
 
     shared = reflection.getSharedObject()
-    if sharePermissions == (OCS_PERMISSION_READ | OCS_PERMISSION_SHARE):
+    if not sharePermissions & OCS_PERMISSION_UPDATE:
       expect_not_modified(os.path.join(d,'TEST_FILE_MODIFIED_USER_SHARE.dat'), shared['md5_sharer'])
```

Comment:
I think this comparison is invalid. The user changed the file locally, I think it should still be changed.
However what should not be changed is the file for the sharer.

My suggestion is to check that the sharer's file is unchanged:
```diff
diff --git a/lib/oc-tests/test_shareFile.py b/lib/oc-tests/test_shareFile.py
index 0731a1a..4a43f45 100644
--- a/lib/oc-tests/test_shareFile.py
+++ b/lib/oc-tests/test_shareFile.py
@@ -125,6 +125,7 @@ def sharer(step):
     shared['TEST_FILE_USER_SHARE'] = share_file_with_user ('TEST_FILE_USER_SHARE.dat', user1, user2, **kwargs)
     shared['TEST_FILE_USER_RESHARE'] = share_file_with_user ('TEST_FILE_USER_RESHARE.dat', user1, user2, **kwargs)
     shared['TEST_FILE_MODIFIED_USER_SHARE'] = share_file_with_user ('TEST_FILE_MODIFIED_USER_SHARE.dat', user1, user
+    shared['sharer.TEST_FILE_MODIFIED_USER_SHARE'] = os.path.join(d,'TEST_FILE_MODIFIED_USER_SHARE.dat')
 
     step (7, 'Sharer validates modified file')
     run_ocsync(d,user_num=1)
@@ -177,7 +178,7 @@ def shareeOne(step):
 
     shared = reflection.getSharedObject()
     if not sharePermissions & OCS_PERMISSION_UPDATE:
-      expect_not_modified(os.path.join(d,'TEST_FILE_MODIFIED_USER_SHARE.dat'), shared['md5_sharer'])
+      expect_not_modified(shared['sharer.TEST_FILE_MODIFIED_USER_SHARE'], shared['md5_sharer'])
 
     step (8, 'Sharee One share files with user 3')
 
```

@PVince81 can you confirm the "sharing" behaviour behind this?